common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/WebViewAuthorizationFragment.java

```diff
@@ -115,13 +117,26 @@ public class WebViewAuthorizationFragment extends AuthorizationFragment {
 
     private PermissionRequest mCameraPermissionRequest;
 
+    // This is used by LegacyFido2ApiManager to launch a PendingIntent received by the legacy API.
+    private ActivityResultLauncher<LegacyFido2ApiObject> mFidoLauncher;
+
     @Override
     public void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        final String methodTag = TAG + ":onCreate";
         final FragmentActivity activity = getActivity();
         if (activity != null) {
             WebViewUtil.setDataDirectorySuffix(activity.getApplicationContext());
         }
+        if (CommonFlightsManager.INSTANCE.getFlightsProvider().isFlightEnabled(CommonFlight.ENABLE_LEGACY_FIDO_SECURITY_KEY_LOGIC)
+                && Build.VERSION.SDK_INT < Build.VERSION_CODES.UPSIDE_DOWN_CAKE) {
+            mFidoLauncher = registerForActivityResult(
+                    new LegacyFidoActivityResultContract(),
+                    result -> {
+                        Logger.info(methodTag, "Legacy FIDO2 API result received.");
+                    }
+            );
```

Comment:
Can you register and manage launcher entirely in LegacyFido2ApiManager? .e.g call in registerForActivityResult(
                    new LegacyFidoActivityResultContract(),
                    result -> {
                        Logger.info(methodTag, "Legacy FIDO2 API result received.");
                    }

in LegacyFido2ApiManager construction